lib/typed_ecto_schema/ecto_type_mapper.ex

```diff
@@ -46,8 +46,13 @@ defmodule TypedEctoSchema.EctoTypeMapper do
     |> add_nil_if_nullable(field_is_nullable?(nullable_default, function_name, opts))
   end
 
-  # Gets the base type for a given Ecto.Type.t()
-  @spec base_type_for(Ecto.Type.t(), field_options()) :: Macro.t()
+  # Gets the base type for a given Ecto.Type.t() or an AST representing a referenced type
+  @spec base_type_for(Ecto.Type.t() | {String.t(), Ecto.Type.t()} | Macro.t(), field_options()) ::
```

Comment:
The `{String.t(), Ecto.Type.t()}` is for the assoc case with custom source, right?
Do you think this type makes sense? Now that I think it is actually a `Ecto.Queryable.t()` isn't it? (which is sad because it is just a `term()`)